### PR TITLE
Refactor bundle for using sonata easy extend

### PR DESCRIPTION
**Pull request by malavsky**

### CODE REVIEW

File: Entity/BaseContact.php [hunk 1]
```diff
@@ -39,21 +46,17 @@ class Contact
...
+        $this->emails = new ArrayCollection();
+        $this->phones = new ArrayCollection();
     }
 
     /**
      * Set name
      *
      * @param string $name
-     * @return Contact
+     * @return BaseContact
```
Comment (aivus) [inline Entity/BaseContact.php]: Please use `return $this` here and in another places

File: Entity/BaseContact.php [hunk 2]
```diff
@@ -169,6 +226,6 @@ public function getUpdatedAt()
      */
     public function __toString()
     {
-        return $this->getName() ?: "Новый контакт";
+        return $this->getName() ?: "New Contact";
```
Comment (aivus) [inline Entity/BaseContact.php]: Please, use single quotes

File: Entity/BasePhone.php [hunk 3]
```diff
@@ -142,6 +151,6 @@ public function getUpdatedAt()
      */
     public function __toString()
     {
-        return $this->getPhone() ?: "Новый телефон";
+        return $this->getPhone() ?: "New Phone";
```
Comment (aivus) [inline Entity/BasePhone.php]: Please, use single quotes

File: Entity/EntityManager/PhoneManager.php [hunk 4]
```diff
@@ -3,20 +3,32 @@
...
+     */
     private $objectManager;
 
-    public function __construct(ObjectManager $objectManager)
+    public function __construct(ObjectManager $objectManager, $class)
     {
         $this->objectManager = $objectManager;
-        $this->repository    = $objectManager->getRepository('GrossumContactBundle:Phone');
+        $this->class = $class;
     }
```
Comment (aivus) [inline Entity/EntityManager/PhoneManager.php:37]: Please, add `getRepository` method

File: Resources/config/admin.yml [hunk 5]
```diff
@@ -1,33 +1,27 @@
...
-            - { name: sonata.admin, manager_type: orm, group: group.hidden, label: "Телефоны" }
+            - { name: sonata.admin, manager_type: orm, group: group.hidden, label: "grossum_contact.admin.phone.plural_label" }
         arguments:
             - ~
             - %grossum_contact.phone.entity.class%
-            - ~
-        calls:
-            - [ setPhoneManager, [@grossum_contact.phone.manager]]
\ No newline at end of file
+            - ~
```
Comment (aivus) [inline Resources/config/admin.yml]: Add new line

File: Resources/config/services.yml [hunk 6]
```diff
@@ -3,13 +3,16 @@ services:
...
             - @grossum_core.entity_manager
+            - %grossum_contact.email.entity.class%
 
     grossum_contact.phone.manager:
         class: %grossum_contact.phone.manager.class%
         arguments:
-            - @grossum_core.entity_manager
\ No newline at end of file
+            - @grossum_core.entity_manager
+            - %grossum_contact.phone.entity.class%
```
Comment (aivus) [inline Resources/config/services.yml]: Add new line

File: Entity/EntityManager/EmailManager.php [hunk 7]
```diff
@@ -3,20 +3,32 @@
...
+     * @var ObjectManager
+     */
     private $objectManager;
 
-    public function __construct(ObjectManager $objectManager)
+    public function __construct(ObjectManager $objectManager, $class)
     {
         $this->objectManager = $objectManager;
-        $this->repository    = $objectManager->getRepository('GrossumContactBundle:Email');
+        $this->class = $class;
```
Comment (aivus) [inline Entity/EntityManager/EmailManager.php:36]: `getRepository` method is missed

Comment (malavsky) [inline Entity/EntityManager/EmailManager.php:36]: fixed

File: Resources/config/services.yml (hunk 6 above)
Comment (malavsky) [inline Resources/config/services.yml]: fixed

File: Admin/EmailAdmin.php [hunk 8]
```diff
@@ -2,75 +2,39 @@
...
 class EmailAdmin extends Admin
 {
-    /** @var  EmailManager $emailManager */
-    private $emailManager;
-
     /**
-     * Fields to be shown on create/edit forms
-     *
-     * @param FormMapper $formMapper
+     * {@inheritdocå}
```
Comment (aivus) [inline Admin/EmailAdmin.php]: Please, remove last char. `å`

File: Admin/PhoneAdmin.php [hunk 9]
```diff
@@ -2,37 +2,25 @@
...
     /**
-     * Fields to be shown on create/edit forms
-     *
-     * @param FormMapper $formMapper
+     * {@inheritdoc}
      */
     protected function configureFormFields(FormMapper $formMapper)
     {
         $formMapper
             ->add('phone', null, ['label' => 'Телефон'])
```
Comment (aivus) [inline Admin/PhoneAdmin.php]: Translation?

File: Admin/PhoneAdmin.php [hunk 10]
```diff
@@ -2,37 +2,25 @@
...
-            ->add('contact', null, ['label' => 'Контакт'])
             ->add('enabled', null, ['label' => 'Включен', 'required' => false]);
```
Comment (aivus) [inline Admin/PhoneAdmin.php]: Translation?

File: Admin/PhoneAdmin.php [hunk 11]
```diff
@@ -41,36 +29,12 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
     /**
-     * Fields to be shown on lists
-     *
-     * @param ListMapper $listMapper
+     * {@inheritdoc}
      */
     protected function configureListFields(ListMapper $listMapper)
     {
         $listMapper
             ->addIdentifier('phone', null, ['label' => 'Телефон'])
```
Comment (aivus) [inline Admin/PhoneAdmin.php]: Translation?

File: Admin/PhoneAdmin.php [hunk 12]
```diff
@@ -41,36 +29,12 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
-            ->add('contact', null, ['label' => 'Контакт'])
             ->add('enabled', null, ['label' => 'Включен']);
```
Comment (aivus) [inline Admin/PhoneAdmin.php]: Translation?

File: DependencyInjection/GrossumContactExtension.php [hunk 13]
```diff
@@ -2,6 +2,7 @@
 
 namespace Grossum\ContactBundle\DependencyInjection;
 
+use Sonata\EasyExtendsBundle\Mapper\DoctrineCollector;
```
Comment (aivus) [inline DependencyInjection/GrossumContactExtension.php]: Please, separate uses

File: Entity/BaseEmail.php [hunk 14]
```diff
@@ -69,10 +59,29 @@ public function getEmail()
...
+    {
+        $this->contact = $contact;
+
+        return $this;
+    }
+
+    /**
+     * @return BaseContact
+     */
+    protected function getContact()
```
Comment (aivus) [inline Entity/BaseEmail.php]: `public`

File: Entity/BasePhone.php [hunk 15]
```diff
@@ -92,10 +82,29 @@ public function getEnabled()
...
+    {
+        $this->contact = $contact;
+
+        return $this;
+    }
+
+    /**
+     * @return BaseContact
+     */
+    protected function getContact()
```
Comment (aivus) [inline Entity/BasePhone.php]: `public`

**Pull request merged**